### ISSUE TITLE
Get weekly offers

### ISSUE THREAD

**Issue by duncte123** [bartmachielsen/SupermarktConnector#6]
I was wondering if it was possible for you to implement a way of getting the current offers/bonus for the stores.

**Comment by bartmachielsen** [comment]
Hi @duncte123 You are talking about the personal offers or the general albert heijn bonus discounts?

**Comment by duncte123** [comment]
I'm talking about the general discounts.
I am aware that you need to log in with a real account for the personal offers

**Comment by bartmachielsen** [comment]
Yes the general discounts are certainly possible, will add it to tasklist. 

The personal offers would also be possible but very difficult, mainly because the offers are based on order patterns, so creating multiple accounts automated will not suffice.

**Comment by talhakosen** [comment]
Does AH, Jumbo allowed for this, Is it legal?

**Comment by bartmachielsen** [comment]
@talhakosen This is an unofficial API, I'm not sure if the Jumbo or Albert Heijn approve this. As far as I know it mostly depends on how you use the data (non-commercial) and that you don't mistreat the server. But I'm not a lawyer.

**Comment by duncte123** [comment]
I did some digging and the endpoint seems to be `mobile-services/discover/v1/bonus`

**Comment by sander102907** [comment]
@duncte123 Hi, I am building an app to suggest me recipes where I want to cross-reference the ingredients of the recipes with the bonus products. The endpoint `mobile-services/discover/v2/bonus` seems to give a subset of all of the bonus products, and also it mostly gives bonus groups instead of the individual products. Is there a way to get all the individual bonus products? Also, how do you find these endpoints?

**Comment by duncte123** [comment]
> @duncte123 Hi, I am building an app to suggest me recipes where I want to cross-reference the ingredients of the recipes with the bonus products. The endpoint `mobile-services/discover/v2/bonus` seems to give a subset of all of the bonus products, and also it mostly gives bonus groups instead of the individual products. Is there a way to get all the individual bonus products? Also, how do you find these endpoints?

I found them by decompiling the app and sifting through all the obfuscated source code for a couple of days

**Comment by bartmachielsen** [comment]
I have implemented the weekly bonus offers in version 0.8:

```
    connector = AHConnector()
    for bonus_prod in connector.get_all_bonus_products():
        print(bonus_prod)
```

If you have any questions, please let me know.
